src/tracer.h

```diff
@@ -157,7 +167,38 @@ class TraceManager {
 
     uint64_t trace_id_;
 
+    // Capture a timestamp generated outside of triton and associate it
+    // with this trace.
+    void CaptureTimestamp(const std::string& name, uint64_t timestamp_ns);
+
 #if !defined(_WIN32) && defined(TRITON_ENABLE_TRACING)
+    /// Initializes Opentelemetry exporter, processor, provider and context.
+    ///
+    /// \param config_map A config map, which stores all parameters, specified
+    /// by user.
+    void InitTracer(const TraceConfigMap& config_map);
+
+    /// Reports TRITONSERVER_InferenceTraceActivity as event to
+    /// the currently active span. If activity is an instance of
+    /// `TRITONSERVER_TRACE_REQUEST_START` or
+    /// `TRITONSERVER_TRACE_COMPUTE_START`,
+    /// it starts a new request or compute span. For the request span it
+    /// adds some triton related attributes, and adds this span to
+    /// `otel_context_`. Alternatively, if activity is
+    /// `TRITONSERVER_TRACE_REQUEST_END` or `TRITONSERVER_TRACE_COMPUTE_END`,
+    /// it ends the corresponding span.
+    ///
+    /// \param trace TRITONSERVER_InferenceTrace instance.
+    /// \param activity  Trace activity.
+    /// \param timestamp_ns Steady timestamp, which is used to calculate
+    /// OpenTelemetry SystemTimestamp to display span on a timeline, and
+    /// OpenTelemetry SteadyTimestamp to calculate the duration on the span
+    /// with better precision.
+    void ReportToOpenTelemetry(
+        TRITONSERVER_InferenceTrace* trace,
+        TRITONSERVER_InferenceTraceActivity activity, uint64_t timestamp_ns);
+
+   private:
```

Comment:
I think that `Trace` is complicated enough to be declare outside `TraceManager` and promote to proper class

Reply:
I'll move it during refactoring in a separate PR